docs/devel/luns.md

```diff
@@ -0,0 +1,82 @@
+This feature intend to support oVirt only.
+
+Design
+------
+
+The requirements for the LUN migration code are (not in priority order)
+
+1. Gather the LUN information from the source destination.
+2. To ease on the user - less manual actions, more automated migration.
+3. Avoid unnecessary risks.
+
+Requirements
+----------------------------------------
+
+1. Destination (OCP cluster) reachability to the ISCSI storage, where the LUN is.
+2. RBAC permission to create `PersistentVolumes` by the controller.
+   - Will be added to the operator manifest.
+
+Risks and Mitigations
+----------------------------------------
+
+1. A VM that will be used while or after the migration with a direct LUN connected.
+   - A warning should be shown before starting the migration.
+   - Detaching the LUN from source VM, once the migration completes.
+   - (Optional) UI checkbox to the user, before starting the migration.
+   - The migration _must_ be a cold migration.
+2. An oVirt version with unsupported API. The [PR](https://github.com/oVirt/ovirt-engine/pull/559) provides LUN details is required.
+   - We may check for missing LUN details and skip the disk migration.
+   - (Optional) Have a mechanism to check the oVirt version for future work.
+3. Creating a `PersistentVolume` requires the file system type: `ext4` or `xfs` of the LUN, which we don't know.
+   - Might not be required for `Block` type.
+4. On non-local destination, the controller needs permission to create the `PersistentVolumes`.
+   - It may be enough to be added to the forklift-operator.
+
+Implementation
+----------------------------------------
+
+The controller gather data from source provider and saves them to the inventory.
+As part of the data gathering we will gather the LUN disks data. We have the VMs disk attachments where we follow to the disks.
+Within the disks API, if the _storage\_type_ is `LUN`, we will gather from the API endpoint `api/disks/<disk_id>` the size, address, IQN and the mapping of the LUN.
+Those will be found withing the `logical_unit` element of the `lun_storage`.
+On the migration plan a new steps will be added, creating a `PersistentVolume`(PV) and a `PersistentVolumeClaim`(PVC) per LUN disk.
+Both of them will contain the relevant data to use the LUN, the PVC will be bound the the PV by a unique label, `<vm_id>-<vm_name>-vol-<volume-number>`.
+When the VM specification created, the controller will add a LUN disk using the PVC that was created as [described by kubevirt](https://kubevirt.io/user-guide/virtual_machines/disks_and_volumes/#lun).
+
+Relevant validations will be performed to know we have the LUN details:
+
+- Allow `LUN` storage type on existing validation.
+- LUN details exist on the provider.
+- SCSI persistent reservation (exists), may required further warning when LUN is detected.
+- (Optional) Make sure the VM is in `down` on source provider(might be not required and be part of the execution plan).
```

Comment:
seems related to the part above which limits this to cold migrations, I can't see why

Reply:
explained on the other part.